olive/passes/pytorch/sparsegpt.py

```diff
@@ -0,0 +1,197 @@
+            ),
+            "percdamp": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Percentage of the average Hessian diagonal to use for dampening. Must be in [0,1].",
+            ),
+            "min_layer": PassConfigParam(
+                type_=int, default_value=None, description="Prune all layers with id >= min_layer."
+            ),
+            "max_layer": PassConfigParam(
+                type_=int, default_value=None, description="Prune all layers with id < max_layer."
+            ),
+            "layer_name_filter": PassConfigParam(
+                type_=Union[str, List[str]],
+                default_value=None,
+                description="Only prune layers whose name contains the given string(s).",
+            ),
+            # this is not the same as accelerator_spec.device which is the target device for inference
+            # compute_device is the device we want to run the algorithm on, does not affect the final model
+            # so accelerator_spec.device can be cpu but compute_device can be cuda for faster pass execution
+            "compute_device": PassConfigParam(
+                type_=str,
+                default_value="auto",
+                description=(
+                    "Device to use for performing computations. Can be 'auto, 'cpu', 'cuda', 'cuda:0', etc. If 'auto',"
+                    " will use cuda if available. Does not affect the final model."
+                ),
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModel, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> PyTorchModel:
+        model_config = model.get_model_config()
+        model_type = model_config.model_type
+        if model_type not in layers_map:
+            raise ValueError(f"Unsupported model type: {model_type}. Supported types: {layers_map.keys()}")
+
+        # get sparsity mode and parameters
+        if isinstance(config["sparsity"], float):
+            assert 0 <= config["sparsity"] <= 1, "Sparsity must be in [0,1]."
+        elif isinstance(config["sparsity"], list):
+            assert len(config["sparsity"]) == 2, "Sparsity must be a float or a list of two integers."
+        mode = "unstructured" if isinstance(config["sparsity"], float) else "structured"
+        sparsity = config["sparsity"]
+        n, m = sparsity if mode == "structured" else [0, 0]
+        logger.debug(f"Running SparseGPT with on {model_type} with mode: {mode}, sparsity: {sparsity}")
+
+        # get device to use for computations
+        device = config["compute_device"]
+        if device == "auto":
+            device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.debug(f"Running SparseGPT on device: {device}")
+
+        # load_data
+        assert config["data_config"] is not None, "Data config is required for SparseGPT pass."
+        logger.debug("Loading data...")
```

Comment:
remove line 105 since 101 is already there.

Reply:
Done!